src/Evolution/Initialization/InterfaceNewtonianEuler.hpp

```diff
@@ -0,0 +1,103 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <utility>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "Evolution/Systems/NewtonianEuler/Tags.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace Initialization {
+
+/// \brief Initialize items related to the interfaces between Elements and on
+/// external boundaries
+///
+/// DataBox changes:
+/// - Adds:
+///   * `face_tags<Tags::InternalDirections<Dim>>`
+///   * `face_tags<Tags::BoundaryDirectionsInterior<Dim>>`
+///   * `face_tags<Tags::BoundaryDirectionsExterior<Dim>>`
+///
+/// - For face_tags:
+///   * `Tags::InterfaceComputeItem<Directions, Tags::Direction<Dim>>`
+///   * `Tags::InterfaceComputeItem<Directions, Tags::InterfaceMesh<Dim>>`
+///   * `Tags::Slice<Directions, typename System::variables_tag>`
+///   * `Tags::InterfaceComputeItem<Directions,
+///                                Tags::UnnormalizedFaceNormal<Dim>>`
+///   * Tags::InterfaceComputeItem<Directions,
+///                                typename System::template magnitude_tag<
+///                                    Tags::UnnormalizedFaceNormal<Dim>>>
+///   * `Tags::InterfaceComputeItem<
+///         Directions, Tags::Normalized<Tags::UnnormalizedFaceNormal<Dim>>>`
+/// - Removes: nothing
+/// - Modifies: nothing
+template <typename System>
+struct Interface {
```

Comment:
Just make the generic one allow you to add compute face and ext tags, and while it's being changed it also should only add prims if the system has prims.

Reply:
Yes, that's what I wanted too, but I'm afraid I'll need a more detailed explanation on how to do this :) Otherwise, I'll move this file to the NE directory.

Reply:
Okay, maybe we can talk about this in person tomorrow (Wednesday)?

Reply:
Here are the metafunctions and code that allows you to pass in a typelist of tags where the first tag will be substituted in with the `Direction` tag.
```cpp
template <typename DirectionsTag, typename VarsTag>
struct Slice {};

template <typename T>
struct SoundSpeed {};

template <template <class...> class F, typename... Ts>
struct interface_holder {
  template <typename Dir>
  using f = F<Dir, Ts...>;
};

template <typename InterfaceHolder, typename Direction>
struct add_direction_tag {
  using type = typename InterfaceHolder::template f<Direction>;
};

using tags = tmpl::list<interface_holder<Slice, SoundSpeed<double>>>;

struct Dir {};

using result = tmpl::transform<tags, add_direction_tag<tmpl::_1, Dir>>;

td<result> aoeu;
```
The compiler output is:
```
test.cpp:167:12: error: implicit instantiation of undefined template
      'td<brigand::list<Slice<Dir, SoundSpeed<double> > > >'
td<result> aoeu;
```
which shows that the transformation was done correctly. The `interface_holder` class can probably just go into the `Initialization` namespace, while the metafunction `add_direction_tag` can go inside the `Initialization::Interface` class.